### ISSUE TITLE
Server-side support for chunked requests

### ISSUE THREAD

**Issue by gterzian** [web-platform-tests/wpt#24139]
`fetch/api/basic/request-upload.any.js` introduces streaming request bodies(using a `ReadableStream`), and it's not clear to me if the server supports these types of requests. 

Also note that the "Transfer-Encoding: chunked" header will be set for such requests, as per Step 4.2 of https://fetch.spec.whatwg.org/#concept-http-network-fetch. 

I was trying this out in Servo, which in theory should pass the test, however I was getting an empty body back in the echo, and noticed the following output:

```
 0:04.54 INFO STDERR: 127.0.0.1 - - [13/Jun/2020 18:22:08] code 400, message Bad request syntax ('4')
 0:04.54 INFO STDERR: 127.0.0.1 - - [13/Jun/2020 18:22:08] "4" 400 -
```

I am not 100% certain that this is a server-side issue, as it might also still be a problem in Servo's client-side code.

**Comment by annevk** [comment]
cc @yutakahirano @yoichio

**Comment by yoichio** [comment]
Yes. I'm seeing exact same response, or empty body with fetch upload streaming feature on Chromium.
I don't know WPT server implementation. What component is supposed to update?

**Comment by gterzian** [comment]
> What component is supposed to update?

I believe the code is in `/tools/wptserve`

**Comment by gterzian** [comment]
Ok so looking at https://en.wikipedia.org/wiki/Chunked_transfer_encoding#Encoded_data

I'm pretty sure the `4` is the first line, indicating a four bytes chunk, in our case the "Test" body sent via the `ReadableStream`. 

So I think on my side, the library I was using to do this is using chunked encoding by default when using their "stream" API(which I plugged into the `ReadableStream`), and the server doesn't know how to handle it. 

Since wptserver is [based](https://github.com/web-platform-tests/wpt/blob/ef14c5bc0da3d3c0f642a3195a9e41433d36a5ba/tools/wptserve/wptserve/server.py#L40) on the Python(2?) stdlib server, I don't think it would support this out of the box. Python 3 also doesn't seem to come with a server supporting this out of the box(although it does come with a [client](https://docs.python.org/3/library/http.client.html#httpconnection-objects) that supports it).

**Comment by stephenmcgruer** [comment]
At this point in time, we do not support chunked encoding. I believe doing so would be a non-trivial amount of work, though we would likely welcome outside contributions towards such a goal (note that a change of that magnitude would likely have to go through the [RFC process](https://github.com/web-platform-tests/rfcs) first, essentially an initial design step).

If someone were to look towards implementing this support, it's important to consider what features tests would need out of the server, which can differ from what clients of a 'normal' server need. For example, we aim to enable tests to have the server do non-spec-compliant things, specifically so that tests can verify what browsers do in those error cases. So it could potentially be even trickier than implementing compliant chunked request support! :)

**Comment by annevk** [comment]
Note that we only need to do this if we decide that we want H/1 support for request body streams, which isn't a settled discussion. And if we support request body streams, the browser would be responsible for the framing, which means the server can be as strict as can be I think as it only needs to consume conforming data and explicitly reject/fail on non-conforming data.
